boot.php

```diff
@@ -1584,3 +1585,50 @@ function infinite_scroll_data($module)
 		return $arr;
 	}
 }
+
+/**
+ * @brief checks if there is a autoconfig.php file in the config folder and automatically installs friendica in case of
+ *
+ * In an automatic setup, autoconfig.php could be created under the "config/" folder with default values
+ * for the mysql instance. At the end, this file should get deleted.
+ *
+ * @return true if the auto-install was succesfully, otherwise false
+ */
+function run_auto_install()
+{
+	$old_htconfig = ((file_exists('.htconfig.php') && filesize('.htconfig.php')) ? true : false);
+
+	if ($old_htconfig) {
+		return rename('.htconfig.php', 'config/.htconfig.php');
```

Comment:
I added this line to automatically move `.htconfig.php` to the `config/` directory.

I tried it at my local instance and it worked for me :-)

Reply:
Ideally, you should create a new post-update function in `update.php`. This would require you to increment DB_UPDATE_VERSION here:

https://github.com/friendica/friendica/blob/0334e49d8e491e83267ad2e6b24a6061713e2b9e/boot.php#L44

Then you take the new value (`1260` at the moment) and you create a `update_1260()` function that performs exactly this.

Reply:
But isn't update.sh just for db-upgrades, is it? It would be the first non-db function.

Just to be sure :-)

Reply:
Indeed, but this is a major breaking change so I don't think we can just issue upgrade notes and everything would go smoothly everywhere, mainly because the `.htconfig.php` is the one file we told people they had to move between install folders when upgrading via an archive.

Reply:
https://github.com/friendica/friendica/blob/0334e49d8e491e83267ad2e6b24a6061713e2b9e/index.php#L190

The problem is that `check_db()` is called in case we are not in mainetenance or install mode. But with the lack of the `.htconfig.php` we **are** in the install mode so friendica shows the installation wizzard.

Shall I change this behavior? I'm worrying that this will get a little bit to far because we had to change friendica-core-functions (f.e. index.php and update_db).

What do you think?

Reply:
Fair enough, maybe we can get away with explicit patch notes. I've been meaning to move the config in a subfolder anyway.

Reply:
So we let the auto-update function in the `autoinstall()` function? Or do you know a better place?

Reply:
We don't move the file automatically, we let admins to it during the upgrade.

Reply:
OK, hopefully they **really** read the patch note..

Reply:
OK, hopefully they **really** read the patch notes..